src/components/Table/GridTable.tsx

```diff
@@ -242,16 +268,32 @@ export function GridTable<R extends Kinded, X extends Only<GridTableXss, X> = {}
     return rows;
   }, [columns, rows, sortOn, sortState, caseSensitive]);
 
-  const hasTotalsRow = rows.some((row) => row.id === "totals");
+  const hasTotalsRow = rows.some((row) => row.id === TOTALS);
+  const hasExpandableHeader = rows.some((row) => row.id === EXPANDABLE_HEADER);
```

Comment:
Question: These are calc-d outside of the `useMemo` but aren't part of the deps array? Should we just calc them inside the useMemo?

Reply:
Ahh, yup. Forgot to add `hasExpandableHeader` to that dep array. Will move this as you suggest, no need to have them defined out here.